### PR TITLE
Add MoveCardAfterReview and MoveCardToReview

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Tested on these two PRs:
https://github.com/pavel-mikula-sonarsource/GitHubActionPlayground/pull/49 - standalone
https://github.com/pavel-mikula-sonarsource/GitHubActionPlayground/pull/52 - fixing issue(s)

### CODE REVIEW

File: CreateCardForStandalonePR/CreateCardForStandalonePR.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-        const pr = this.payload.pull_request as WebhookPayload["pull_request"] & { id: number };
-        const matches = pr.body?.match(/(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s*#\d+/gi);
-        if (matches) {
-            this.log(`Skip, contains '$(matches[0])'`);
+        const pr = this.payload.pull_request as components["schemas"]["issue"];
+        const fixedIssues = this.fixedIssues(pr);
+        if (fixedIssues) {
+            this.log(`Skip, fixes issues`);
         } else {
             this.addAssignee(pr, this.payload.sender.login);
```
Comment (costin-zaharia-sonarsource) [inline CreateCardForStandalonePR/CreateCardForStandalonePR.ts:11]: Missing await.

File: lib/ProjectContent.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    private readonly columns: number[] = [];
+
+    private constructor(action: OctokitAction, columns: number[]) {
+        this.action = action;
+        this.columns = columns;
+    }
+
+    public static async FromColumn(action: OctokitAction, column_id: number): Promise<ProjectContent> {
+        const { data: column } = await action.rest.projects.getColumn({ column_id });
+        const project_id = parseInt(column.project_url.split("/").pop());
```
Comment (costin-zaharia-sonarsource) [inline lib/ProjectContent.ts:17]: I wonder if pop will create a copy of the array (it does change the length) and how this affects performance. 

Probably it's not a problem for this script that runs once but would be nice to know.

File: lib/ProjectContent.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        const { data: columns } = await action.rest.projects.listColumns({ project_id }); 
+        return new ProjectContent(action, columns.map(x => x.id));
+    }
+
+    public async findCard(issueOrPR: IssueOrPR): Promise<any> {
+        // We should start caching these results in case we need to call it multiple times from a single action
```
Comment (costin-zaharia-sonarsource) [inline lib/ProjectContent.ts:23]: Any reason not to do this now?

File: lib/ProjectContent.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        const content_url = issueOrPR.issue_url ?? issueOrPR.url;
+        for (const column_id of this.columns) {
+            const cards = await this.action.rest.projects.listCards({ column_id });
+            const card = cards.data.find(x => x.content_url == content_url);
```
Comment (costin-zaharia-sonarsource) [inline lib/ProjectContent.ts:27]:
```suggestion
            const { data: cards } = await this.action.rest.projects.listCards({ column_id });
            const card = cards.find(x => x.content_url == content_url);
```

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]

File: lib/ProjectContent.ts (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply lib/ProjectContent.ts:17]: I don't even know how to measure that thing :) I'll leave it a mystery, as it doesn't matter here

File: lib/ProjectContent.ts (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply lib/ProjectContent.ts:23]: Not to overengineer it, we don't need it (yet). And I'm not sure if we will any time soon.

File: lib/OctokitAction.ts [hunk 5]
```diff
@@ -29,11 +31,33 @@ export abstract class OctokitAction extends Action {
...
-    protected async downloadData(url: string): Promise<any> {
+    public async downloadData(url: string): Promise<any> {
         console.log("Downloading " + url);
         return (await this.octokit.request(url)).data;
     }
 
+    protected async getIssue(issue_number: number): Promise<components["schemas"]["issue"]> {
+        try {
+            this.log(`Getting issue #${issue_number}`);
+            return (await this.rest.issues.get(this.addRepo({ issue_number }))).data;
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply lib/OctokitAction.ts:42]: I gave it a try to practice it. Victor said that there's nothing wrong with the old implementation, it's only about if we like chaining.

[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply lib/OctokitAction.ts:42]: ..aaand I've reverted the update after one more discussion. Chaining is the old-style way of doing it. async/await is the new syntax sugar on how to do the same thing.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**